src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+class MemcachedAdapter extends AbstractAdapter
+{
+    private $client;
+
+    /**
+     * @param \Memcached $client
+     */
+    public function __construct(\Memcached $client, $namespace = '', $defaultLifetime = 0)
+    {
+        parent::__construct($namespace, $defaultLifetime);
+        $this->client = $client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave(array $values, $lifetime)
+    {
+        return $this->client->setMulti($values, $lifetime)
+            && $this->client->getResultCode() === \Memcached::RES_SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        foreach ($this->client->getMulti($ids) as $id => $val) {
+            yield $id => $val;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id)
+    {
+        return $this->client->get($id) !== false
```

Comment:
That is an error on my part: intended for that to be an OR (||). The idea is to avoid the additional call to `$this->client->getResultCode()` if the `$this->client->get($id)` isn't `false`.